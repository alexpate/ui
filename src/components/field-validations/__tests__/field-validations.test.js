/* @flow */
/* eslint-env jest */
import expect from 'expect';
import {shallow} from 'enzyme';
import {testComponent} from 'utils/test/react';

import FieldValidations from '../field-validations';
import FieldValidation from '../../field-validation';

describe('components/field-validations', () => {
  describe('FieldValidations', () => {
    const create = testComponent(FieldValidations, () => ({
      validations: [
        {
          message: 'Email canâ€™t be empty',
        },
        {
          message: 'Email must be of email type',
        },
      ],
    }));

    it('should render shallow component ok', () => {
      const {element} = create();
      const result = shallow(element);
      expect(result).toExist('should render OK');
    });

    it('should render two validation components', () => {
      const {element} = create();
      const result = shallow(element);
      expect(result.find(FieldValidation).length).toEqual(2);
    });
  });
});
