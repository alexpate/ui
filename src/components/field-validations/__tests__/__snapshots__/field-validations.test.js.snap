// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldValidations should render shallow component ok 1`] = `
<div
  className="css-1ylngii css-1b9obwq0"
/>
`;
